pkgs/initiator/initiator.go

```diff
@@ -616,21 +676,36 @@ func (c *Initiator) StartReshare(newId, oldID [24]byte, oldIDs, newIDs []uint64,
 	if err != nil {
 		return nil, err
 	}
-	dkgResults, _, _, ssvContractOwnerNonceSigShares, err := c.ProcessReshareResultResponse(dkgResult, newId)
+	dkgResults, _, _, _, ssvContractOwnerNonceSigShares, err := c.ProcessDKGResultResponse(dkgResult, newId)
 	if err != nil {
 		return nil, err
 	}
 	c.Logger.Info("🏁 DKG completed, verifying deposit data and ssv payload")
 	// Recover and verify Master Signature for SSV contract owner+nonce
-	reconstructedOwnerNonceMasterSig, err := crypto.RecoverMasterSig(ssvContractOwnerNonceSigShares)
+	reconstructedOwnerNonceMasterSig, err := crypto.RecoverMasterSig(newIDs, ssvContractOwnerNonceSigShares)
 	if err != nil {
 		return nil, err
 	}
-	keyshares, err := GeneratePayload(dkgResults, reconstructedOwnerNonceMasterSig.Serialize())
+	keyshares, err := GeneratePayload(dkgResults, reconstructedOwnerNonceMasterSig.Serialize(), owner, nonce)
 	if err != nil {
 		return nil, err
 	}
 	c.Logger.Info("✅ verified master signature for ssv contract data")
+	// sending back to operators results
+	keysharesData, err := json.Marshal(keyshares)
+	if err != nil {
+		return nil, err
+	}
+	resultMsg := &wire.ResultData{
+		Operators:     newOps,
+		Identifier:    newId,
+		DepositData:   nil,
```

Comment:
so we're not sending the deposit data? only keysharesdata?

Reply:
when resharing we get only a new `keyshares` file, validator pub and deposit stays the same, so no need to send it

Reply:
oh I see this is only for resharing, missed that, thanks.